src/app/marketplace/edit-ad.component.ts

```diff
@@ -227,12 +245,11 @@ export class EditAdComponent
    */
   save(insertNew?: boolean) {
 
-    validateBeforeSubmit(this.form);
-    if (!this.form.valid) {
+    if (!validateBeforeSubmit(this.form)) {
       return;
     }
 
-    const value = this.form.value;
+    const value = cloneDeep(this.form.value);
```

Comment:
The `form.value` can be safely mutated without changing the internal form state. Hence, there's no need for cloneDeep.

Reply:
When doing a delete of any property of the form.value it really mutates and no longer contains it.